### PR TITLE
Fix problem with client timeouts

### PR DESCRIPTION
**Pull request by bryce-anderson**
Previous strategy tried to use the callback from a Task.async multiple times if timeout occurs which is a nogo.

### CODE REVIEW

**Comment by bryce-anderson** [comment]
I have some cleanup I want to include in this. Don't merge.

**Comment by bryce-anderson** [comment]
Ok, this should be good to go.

File: blaze-client/src/main/scala/org/http4s/client/blaze/Main/Main.scala [hunk 1]
```diff
@@ -5,6 +5,7 @@ import org.http4s.Uri
 import org.http4s.EntityDecoder._
 
 import scalaz._
+import scalaz.concurrent.Task
 
 object Main {
```
Comment (rossabaker) [inline blaze-client/src/main/scala/org/http4s/client/blaze/Main/Main.scala]: This is in src/main?

**Comment by rossabaker** [comment]
I wonder whether certain exceptions could be case objects, when they're used more as values.  InProgressException, for instance.  Not really a big deal, but can lead to gains if they're thrown a lot.

**Comment by bryce-anderson** [comment]
Ugg, that is debug code that has somehow leaked into the blaze-client. I'll remove it. ```InProgressException``` should be rare, but you're right, its better used as a case object.

**Comment by bryce-anderson** [comment]
@rossabaker, any other concerns about this PR?

As an aside, I'm a little dissatisfied with the blaze-client architecture in general but its served its purpose: experimental backend. We should consider what more we demand from a backend (like so called client middleware) and then I may spend some time reworking this to be more manageable.

**Pull request merged**